### ISSUE TITLE
Potential regression in v1.23 with respect to docker.host setting

### ISSUE THREAD

**Issue by skilkis** [microsoft/vscode-docker#3802]
I have been using this extension on Windows 10 to access Docker remotely which is running on WSL2. Previously to v1.23 I was able to use the `docker.host` setting to configure the remote port and have containers/images display correctly within the extension. However, the new `DOCKER_HOST` variable in v1.23 doesn't work for me.  This seems to be related to #3783 , although I do not have problems on my system when using the extension when VS Code is launched in WSL. As a quick check, I did verify that a correct response is returned from `http://localhost:2375/images/json` and downgrading to v1.22 of the extension solves this issue.

```json
{
  # Doesn't work with v1.23
  "docker.environment": {
    "DOCKER_HOST": "http://localhost:2375"
  },
  # Works fine on v1.22
  "docker.host": "http://localhost:2375",
}
```

OS: Ubuntu Windows 10 x64 10.0.19045
VSCode Version: 1.74.3
Docker extension version: 1.23.3
Docker version: Docker version 20.10.22, build 3a2c30b

**Comment by bwateratmsft** [comment]
Do you have the Docker CLI installed on the Windows "host", or is it only present in the WSL instance? Right now, with #3263, we're exclusively using the CLI from the context of the extension host, but we are working on enabling it to work through WSL (i.e., instead of running `docker ps ...` we'd do `wsl -- docker ps ...`).

Having the CLI present on the Windows host is one option here that I think should work. Downgrading to 1.22.2 is another option, or you can also check out the [WSL extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl), if having the WSL instance as the development environment works for you.

**Comment by bwateratmsft** [comment]
I just noticed we'd forgotten to open an issue for the aforementioned do-things-in-WSL feature, so I created #3803.

**Comment by skilkis** [comment]
@bwateratmsft thank you for this information! Installing Docker CLI on Windows indeed resolves this issue on 1.23. Would be great to have the ability to run CLI commands from within the WSL instance in the future

**Comment by bwateratmsft** [comment]
I agree! We can track that in #3803, so I'll resolve this one as a dupe of that.
